src/indexes/common.ts

```diff
@@ -6,10 +6,15 @@ export type TargetProperty = string;
 export type IndexOptions<T extends IndexableObj> = {
   targetProperties: Properties;
 };
-export type Extension = {
+export type Extension<T extends IndexableObj> = {
   deleteFromIndex: () => void ;
+  // It looks like Omit<T & Extension<T>, keyof Extension<T>> might just equal T.
+  // But that's not true if T intersects with Extension eg T includes a field called 'deleteFromIndex' or 'getTarget'
+  // We could probably guard against this using a conditional type... but let's not get too carried away
+  getTarget: () => Omit<T & Extension<T>, keyof Extension<T>>;
 };
-export type Captor<T extends IndexableObj> = (obj: T) => T & Extension;
+
+export type Captor<T extends IndexableObj> = (obj: T) => T & Extension<T>;
```

Comment:
```suggestion
export type Captor<T extends IndexableObj> = (obj: T) => Extension<T>;
```

?

Reply:
Yeah sure, I can create an alias for T & Extension<T>?

Reply:
I don't think you need to? 

```ts
export type Extension<T extends IndexableObj> = T & {
  deleteFromIndex: () => void;
  getTarget: () => T;
}
```

that should work along with the changes to `IndexableObj`

Reply:
you might run into the dreaded higher kinded type error, but afaik there is still no away around that one, just have to ts-expect-error it